komodo/link.py

```diff
@@ -0,0 +1,56 @@
+import time
+import os
+import shutil
+
+import komodo
+
+age_limit = 7200
+
+
+def update_links(prefix):
+    # The first step is to scan the prefix directory and inspect all the
+    # directories in the folder. The directory names are split, and a
+    # dictionary:
+    #
+    #     release_list = {"2018.03" : [(1234, "2018.03-1234"),
+    #                                  (5678, "2018-03-5678")],
+    #                     "2018.10" : [(561, "2018.10.561"),
+    #                                  (777, "2018.10-777")]}
+    #
+    # This is dictionary where the release names like '2018.03' and '2018.10'
+    # are used as keys, the values in the dictionary are lists of tuples
+    # (timestamp, path_name).
+    ts_now = time.mktime( time.localtime() )
+    release_list = {}
+    for entry in os.listdir(prefix):
+        if os.path.islink(entry):
+            continue
+
+        if os.path.isdir(entry):
+            release, ts = komodo.split_release_name(entry)
+            if ts:
+                if not release in release_list:
+                    release_list[release] = []
+
+                release_list[release].append((ts, entry))
+
+    for release,dir_list in release_list.items():
+        if os.path.exists(release):
+            if not os.path.islink(release):
+                raise Exception("Sorry - {} already exists - and is not a symblic link. This must be fixed manually".format(release))
```

Comment:
These two `if`s should be `and`ed in one if and the Exception should possibly be IO (and no need to say sorry here)

Reply:
Fixed